### PR TITLE
trying to make an entry list

### PR DESCRIPTION
**Pull request by yaaHosein**
let us to work together

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,19 +3,17 @@ import React from 'react';
...
   const items = props.items;
-  const listItems = numbers.map((item) =>
+  const listItems = items.map((item) =>
     <li>{item}</li>
   );
   function App() {
     return (
       <ul>{listItems}</ul>
     );
   }
```
Comment (SMH110) [inline src/App.js]: Can you explain what this `App` doing? what is the need for this?

Comment (yaaHosein) [reply src/App.js]: As I titled the pull request

Comment (SMH110) [reply src/App.js]: I want more detail

File: src/App.js [hunk 2]
```diff
@@ -3,19 +3,17 @@ import React from 'react';
 import './App.css';
```
Comment (SMH110) [inline src/App.js]: Can you please remove everything in this file and create a simple App component function that returns empty `div`

and make sure to export that component using the export keyword

**Comment by yaaHosein** [comment]
> Why this called `MyList` not `EntryList` ?

This question came in my mind but I thought it is not a big issue so I did not change it

File: src/MyList.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+function MyList() {
```
Comment (SMH110) [inline src/MyList.js]: Do you still need this file?

File: src/App.js [hunk 4]
```diff
@@ -1,21 +1,7 @@
-import React from 'react';
+function App() {
+  return (
+      <div>
```
Comment (SMH110) [inline src/App.js]: You need to call `EntryList` here.
Remember EntryList is a component function so I want you to call it as component function.
I don't want you to do `EntryList()`

File: src/MyList.js (hunk 3 above)
Comment (yaaHosein) [reply src/MyList.js]: when I wanted to change the file name inside EntryList file I could not because of visual code safe mode 
so if you noticed I send you many changes because I cut all codes and paste them in a notepad file 
then I did my new changes then i pasted them to visual code and pushed my changes 

now aslo I want to complete importing but n letter in the keyboard is not working because of the safe mode 
I have to fix this problem then try 

there is no more MyList in my codes I do not know why they are did not removed in GitHub

Comment (yaaHosein) [reply src/MyList.js]: image

Comment (SMH110) [reply src/MyList.js]: click on Manage and Trust this file

Comment (yaaHosein) [reply src/MyList.js]: image

Comment (yaaHosein) [reply src/MyList.js]: as you can see the folder is trusted but still i can not call EntryList file in App.js file

Comment (SMH110) [reply src/MyList.js]: what do you mean I can't call EntryList file in App.js

Comment (yaaHosein) [reply src/MyList.js]: I was trying to to write 
import Entry ....etc 

but I was able to write till import E

after many tires also i could not
so I wrote the call line in a notepad file and copy and paste is in visual studio edit

**Comment by yaaHosein** [comment]
screen shot 2

File: src/App.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import { EntryList } from "./EntryList";
```
Comment (SMH110) [inline src/App.js]: You just have imported the EntryList component
you need to render it inside App component

Comment (yaaHosein) [reply src/App.js]: image

Comment (yaaHosein) [reply src/App.js]: image

Comment (SMH110) [reply src/App.js]: Did you export EntryList ?

Comment (yaaHosein) [reply src/App.js]: image

**Comment by yaaHosein** [comment]
image

should I do anything in Source control

File: src/EntryList.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+function EntryList() {
+
+    return (
+        <div>
+            <label>My List</label>
+            <tr>
+                <th>Enter Something</th>
```
Comment (SMH110) [inline src/EntryList.js]: Please don't add code that you don't know what's is doing

you added tr tag which belongs to table tag

File: src/App.js [hunk 7]
```diff
@@ -1,21 +1,10 @@
-import React from 'react';
+import { EntryList } from "./EntryList";
```
Comment (SMH110) [inline src/App.js]: You have exported EntryList as `default` and in this case you need to import it as default 

like this

import EntryList from "./EntryList" 

note I used `EntryList` here as name but you can use any name you like 

you can do 

`import Foo from "./EntryList"`
or 
`import Abcdef from "./EntryList"`

File: src/EntryList.js (hunk 6 above)
Comment (yaaHosein) [reply src/EntryList.js]: I wanted to add a cod for rows

File: src/App.js (hunk 7 above)
Comment (yaaHosein) [reply src/App.js]: in our previous project :
import { Flowers } from "./Flowers";

it worked for us 
also we had in the end:
export default App;

Comment (yaaHosein) [reply src/App.js]: image

image

Comment (SMH110) [reply src/App.js]: > in our previous project : import { Flowers } from "./Flowers";
> 
> it worked for us also we had in the end: export default App;

Can you check pls how the Flowers is exported in the previous project?
did I use `default` keyword also or just `export` ?

File: src/EntryList.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+function EntryList() {
```
Comment (SMH110) [inline src/EntryList.js]: EntryList should accept `itmes` as props.

`items` is an array of strings.

EntryList should render the content of the `items` array in `span` tag 

Please remove your changes from EntryList  and ensure to follow my instructions 
I mean remove this code 
```html
<div>
            <h6 className="yu" style="color:blue;">My List</h6>
            <form>
                <label>
                    <input type="text" placeholder="Enter Something"> </input>
                </label>
            </form>
            <div class="w3-padding w3-xlarge w3-teal">
                <i class="glyphicon glyphicon-trash"></i>

            </div>
        </div>
```

File: src/App.js (hunk 7 above)
Comment (yaaHosein) [reply src/App.js]: I have scrutinized this point before asking you and I found 
we used :
export default App;

but I did not open Flowers.js file 
today when I had a look no we did not use default

File: src/EntryList.js (hunk 8 above)
Comment (yaaHosein) [reply src/EntryList.js]: please why should ?
I want to see if I can do the job without using props
how did you know the items are strings ?
why not div tag ?

Comment (SMH110) [reply src/EntryList.js]: > please why should ? I want to see if I can do the job without using props how did you know the items are strings ? why not div tag ?

I think you didn't understand the requirement.

I want the EntryList to display list of items 


for example if I have this array of data `["one", "two", "three"]`

I want to be able to use EntryList to display the data above 

in the same time, I want EntryList to be very flexible and can be used to show any list of items 

for example 

it should be able to use to it to show this list also 

- Apples
- Bananas 
- Eggs 

 In short, I want it to be reusable 
what need to be done to the EntryList ?

Comment (yaaHosein) [reply src/EntryList.js]: No problem 
But I wanted first to do the job without using the props 
Later with it
Just I wanted to see what is the difference

Comment (yaaHosein) [reply src/EntryList.js]: You sent me this 
A53195F2-C53B-43BE-A98E-D8C20A8C931E

So how you want the items in your second example 
All the list in one box 
Or each item ( fruit 🍌  🍉 ) in one box ?!

2) why you did not accept table codes and also wanted me to remove html form codes 
So what kind of codes you want 
Thanks

File: src/EntryList.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+    return (
+        <>
+                <span>
+                    <ul>
+                        Apple
+                        watermelon
```
Comment (SMH110) [inline src/EntryList.js]: Not very bad, but here you have hardcoded values "Apple" "watermelon" 
I want you to make Entrylist renders any list without having to touch the code inside EntryList 

Currently EntryList shows "Apple" "watermelon" "banana" always 
for example if  I use it in my project 
`
<EntryList />`
`
<EntryList />`

 "Apple" "watermelon" "banana"
 "Apple" "watermelon" "banana"

I want you to remove the hardcoded values ( "Apple" "watermelon" "banana") from EntryList and passe them using `props`

and to do that you can pass these items ( "Apple" "watermelon" "banana") as props to EntryList 

so in the App component you can do 

`,<EntrlyList items={[ "Apple" "watermelon", "banana"]} />`

then you have to do some changes to EntrlyList to make it render list 
I assume you have came across `[].map` and you should be able to use it to show the list

also for each item in the ist you need `li` tag

Comment (yaaHosein) [reply src/EntryList.js]: I wanted to use map method 
But first I wanted to see what are my mistakes without using map

File: src/App.js [hunk 10]
```diff
@@ -1,21 +1,28 @@
...
+  return (
+    <div className="App">
+      <EntryList fruit="Apple" />
+      <EntryList fruit="Banana" />
+      <EntryList fruit="Orange" />
+    </div>
   );
+
 }
+const items = data.map(function (item) {
```
Comment (SMH110) [inline src/App.js]: This should be removed

Comment (yaaHosein) [reply src/App.js]: As you can see I added data for adding our items 
If I removed the data file where we can add or items names?!

Comment (SMH110) [reply src/App.js]: You can define array of fruits on the App.js 

const fruits = ["Apples", "Oranges"] 

Pass this array as props to `EntryList` 

Make sure that EntryList render each item in the array using `li` tag

File: src/EntryList.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import App from "./App";
```
Comment (SMH110) [inline src/EntryList.js]: Do you need App here? You have imported App?

File: src/EntryList.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+function EntryList(props) {
+    return
+    (
+        <ul>
+            {fruits.map(function (fruit) {
```
Comment (SMH110) [inline src/EntryList.js]: this is correct, but where `fruits` is defined?
there's something missing before `fruits`

File: src/App.js [hunk 13]
```diff
@@ -1,21 +1,15 @@
...
-    return (
-      <ul>{listItems}</ul>
-    );
-  }
-  const items = [item1, item2, item3, item4, item5];
-  ReactDOM.render(
-    <MyList items={items} />,
-    document.getElementById('root')
-  );
+const fruits = ["carrot"]
```
Comment (SMH110) [inline src/App.js]: this is good, you have created fruits array 
this array can be passed to `EntryList`

File: src/App.js [hunk 14]
```diff
@@ -1,21 +1,15 @@
...
+function App() {
+  return
+  (
+    <div className="App">
+      <EntryList fruits={["carrot"]} />
```
Comment (SMH110) [inline src/App.js]: ` <EntryList fruits={["carrot"]} />`

`fruits` is props on `EntryList`

you're not passing the array you created on line 3 
you are passing another array to EntryList 

this can be fixed 

`<EntryList fruits={fruits} />`

File: src/EntryList.js (hunk 11 above)
Comment (yaaHosein) [reply src/EntryList.js]: I do not know 
Just I thought maybe I need it 
Because there are some codes in it belongs to some codes in App.js

File: src/App.js (hunk 13 above)
Comment (yaaHosein) [reply src/App.js]: Okay

File: src/EntryList.js (hunk 11 above)
Comment (SMH110) [reply src/EntryList.js]: No, you don't need it, 
if there's a code in App.js and you need it here, then you should import that code.
You've imported App component which you don't need

File: src/EntryList.js [hunk 15]
```diff
@@ -0,0 +1,12 @@
+import App from "./App";
+function EntryList(props) {
+    return
+    (
+        <ul>
+            let fruits =    {fruits.map(function (fruit) {
```
Comment (SMH110) [inline src/EntryList.js]: This is wrong
where the fruits array should come from?

File: src/App.js [hunk 16]
```diff
@@ -1,21 +1,15 @@
...
-  ReactDOM.render(
-    <MyList items={items} />,
-    document.getElementById('root')
-  );
+const fruits = ["carrot"]
+function App(props) {
+  return
+  (
+    <div className="App">
+      <EntryList fruits={props.fruits} />
```
Comment (SMH110) [inline src/App.js]: this is wrong, if you read my previous comment I gave the solution

File: src/App.js [hunk 17]
```diff
@@ -1,21 +1,15 @@
...
-      <ul>{listItems}</ul>
-    );
-  }
-  const items = [item1, item2, item3, item4, item5];
-  ReactDOM.render(
-    <MyList items={items} />,
-    document.getElementById('root')
-  );
+const fruits = ["carrot"]
+function App(props) {
```
Comment (SMH110) [inline src/App.js]: App component does need props, you're not passing anything to App component

**Comment by yaaHosein** [comment]
do you thing it will good running in this project
npm install

File: src/App.js [hunk 18]
```diff
@@ -1,21 +1,15 @@
...
-  ReactDOM.render(
-    <MyList items={items} />,
-    document.getElementById('root')
-  );
+const fruits = ["carrot"]
+function App() {
+  return
+  (
+    <div className="App">
+      <EntryList fruits={fruits} />
```
Comment (SMH110) [inline src/App.js]: THis is correct

File: src/EntryList.js [hunk 19]
```diff
@@ -0,0 +1,12 @@
+function EntryList(props) {
+
+    return
+    (
+        <ul>
+            let fruits =    {fruits.map(function (fruit) {
```
Comment (SMH110) [inline src/EntryList.js]: Please remove `let fruits =    ` and use the props parameter instead 
`props.fruits`

File: src/EntryList.js [hunk 20]
```diff
@@ -3,8 +3,8 @@ function EntryList(props) {
     return
     (
         <ul>
-            let fruits =    {fruits.map(function (fruit) {
-                return <li>{props.fruit}</li>
+            {fruits.map(function (fruit) {
+                return <li>{props.fruits}</li>
```
Comment (SMH110) [inline src/EntryList.js]: wrong
it should be 
`
{props.fruites.map ...etc }`

File: src/App.js [hunk 21]
```diff
@@ -5,7 +5,7 @@ function App() {
   return
   (
     <div className="App">
-      <EntryList fruits={fruits} />
```
Comment (SMH110) [inline src/App.js]: why did you remove it?
WHY?
